app/controllers/api/subjects_controller.rb

```diff
@@ -0,0 +1,40 @@
+module Api
+  class SubjectsController < ApplicationController
+    include Rails.application.routes.url_helpers
+
+    def index
+      subjects = Subject.all
+
+      # Filter by speciality if provided
+      if params[:speciality_id].present?
+        subjects = subjects.joins(:specialities)
+                           .where(specialities: { id: params[:speciality_id] })
+      end
+
+      # Filter by exam if provided
+      if params[:exam_id].present?
+        subjects = subjects.joins(:specialities)
+                           .where(specialities: { exam_id: params[:exam_id] })
+      end
+
+      # ✅ New: Filter by year
+      if params[:year].present?
+        subjects = subjects.where(year: params[:year].to_i)
+      end
+
+      subjects = subjects.distinct
+
+      render json: subjects.map { |s|
```

Comment:
For the sake of developers sanity, you should strictly avoid one or 2 words variables. They are easily results in bugs and carry no meaning. Name your variable here a singular version of the collection being looped.
It is also preferable to use the `do ... end` instead of `{}` when your loop takes multiple lines.

```suggestion
      render json: subjects.map do |subject|
```